lib/learn_open/services/git_ssh_connector.rb

```diff
@@ -0,0 +1,40 @@
+module LearnOpen
+  class GitSSHConnector
+    attr_reader :ssh_connection, :environment
+
+    GIT_SSH_USER = 'git'
+
+    def self.call(git_server:, environment:)
+      self.new(git_server: git_server, environment: environment).call
+    end
+
+    def initialize(git_server:, environment:)
+      @ssh_connection = LearnOpen.ssh_adapter.new(user: GIT_SSH_USER, hostname: git_server)
+      @environment = environment
+    end
+
+    def call
+      if managed_environment? && ssh_unauthenticated?
+        upload_ssh_keys!
+      end
+
+      ssh_authenticated?
+    end
+
+    def upload_ssh_keys!
+      LearnOpen.learn_web_client.add_ssh_key(key: ssh_connection.public_key)
```

Comment:
This was something I talked to @StevenNunez about. We landed on the idea it would probably be too invasive to push whatever happens to be the public key in `~/.ssh/id_rsa.pub` to a public Github account without explicit consent, though if we ever introduce multiple Enterprise accounts per use then we could potentially fan those keys out.

Reply:
Just thinking ahead, are you also working on how to find the right `GithubAccount` for a user? I don't think I see any info about the server being passed through [learn-web](https://github.com/learn-co/learn-web/pull/6).

Reply:
Ah interesting. I'll think about that.

Reply:
I was thinking we push it to all of your accounts.